src/resolvers/contextual/workspace-resolver.js

```diff
@@ -19,19 +19,36 @@ export default class WorkspaceResolver extends BaseResolver {
 
   workspaceLayout: WorkspaceLayout;
 
-  resolve(): Promise<Manifest> {
+  resolve(downloadedManifest: ?Manifest): Promise<Manifest> {
     const workspace = this.workspaceLayout.getManifestByPattern(this.request.pattern);
     invariant(workspace, 'expected workspace');
     const {manifest, loc} = workspace;
+    if(manifest._remote && manifest._remote.registryRemote){
+      return Promise.resolve(manifest); //already downloaded
+    }
     const registry = manifest._registry;
     invariant(registry, 'expected reference');
-
-    manifest._remote = {
+    let hash = '', registryRemote;
+    if(downloadedManifest && manifest.version === downloadedManifest.version){
+      registryRemote = downloadedManifest._remote;
+      hash = registryRemote.hash;
+      //override any local changes to manifest
+      Object.keys(manifest).forEach(k => k.startsWith('_') || delete manifest[k]);
```

Comment:
this was to avoid weird bugs where the local package.json differed from the remote one (for the same version).